doc/concepts/moveit_task_constructor/moveit_task_constructor.rst

```diff
@@ -0,0 +1,212 @@
+.. _MoveIt Task Constructor:
+
+#######################
+MoveIt Task Constructor
+#######################
+
+What is MoveIt Task Constructor?
+--------------------------------
+
+| The MoveIt Task Constructor framework helps break down complex planning tasks to multiple interdependent subtasks.
+| The MTC framework uses MoveIt to solve the subtasks and a common interface, based on MoveIt's PlanningScene, is used to pass solution hypotheses between stages.
+
+MTC Stages
+-----------
+| A MTC stage refers to a component or step in the task execution pipeline.
+| Stages can be arranged in any arbitrary order and its hierarchy is only limited by the individual stages types.
+| The order in which stages can be arranged is restricted by the direction in which results are passed.
+
+There are three possible stages relating to the result flow:
+
+* Generators
+
+* Propagators
+
+* Connectors
+
+Generator Stage
+^^^^^^^^^^^^^^^
+| Generators compute results and pass them in both direction - upwards and downwards.
+| Execution of a MTC task starts with the Generator stages.
+| The most important generator stage is `CurrentState`, which provides the current robot state as the starting point for a planning pipeline.
+
+| Monitoring Generator - Generator that monitors the solution of another stage to make reuse of them.
+| Main example of Monitoring Generator - `GeneratePose`. It usually monitors a `CurrentState` or `ModifyPlanning Scene` stage.
+
+Currently available generator stages:
+
+* CurrentState
+
+* FixedState
+
+* Monitoring Generators - GeneratePose, GenerateGraspPose, GeneratePlacePose, GenerateRandomPose
+
+| Link for more information on generator stages available to use :ref:`Generating Stages`.
+
+Propagating Stage
+^^^^^^^^^^^^^^^^^
+| Propagators receive the result from one neighbor state, solve a subproblem and then propagate the result to the neighbor on the opposite side.
+| Depending on the implementation, this stage can pass solutions forward, backward or in both directions.
```

Comment:
```suggestion
| Depending on the implementation, this stage can pass solutions forward, backward, or in both directions.
```

Also, this paragraph says forward/backward whereas the generators say upwards/downwards. Which one should we use consistently?